### PR TITLE
track and disconnect audio resources used by setSinkId

### PR DESCRIPTION
**Pull request by kross-italk**
Short description of changes:

Invoking setSinkId will create an Audio and MediaStream.
Create them once per wavesurfer and then disconnect/NULL them explicitly
on destroy.

After this change observe chrome://media-internals and create/setSinkId/destroy
many wavesurfers in a loop

Related Issues and other PRs:
fixes https://github.com/katspaugh/wavesurfer.js/issues/2481

Demo of media-internals after this change:
kross-2481-media-internals

### CODE REVIEW

File: src/webaudio.js
Comment (msuhov) [inline src/webaudio.js:309]: Hi @kross-italk! Thanks for your commit

Quick suggestion: since we already have `this.audio` variable, I don't think we need to reassign it to another var, just use it directly in this method

File: CHANGES.md
Comment (thijstriemstra) [reply CHANGES.md:6]: and no need to add a release date (use `unreleased` instead)

File: CHANGES.md
Comment (thijstriemstra) [inline CHANGES.md:7]: ticket nr should go at end of sentence (see other changelog entries)

File: src/webaudio.js
Comment (thijstriemstra) [inline src/webaudio.js:149]: i don't like `dest` as a var name on a class level. Can you call it something like `sinkDestination` or `sinkStreamDestination`? Same goes for `this.audio`, something like `sinkAudioElement`?

File: src/webaudio.js
Comment (thijstriemstra) [inline src/webaudio.js:552]: add a comment making it clear this is related to `setSinkId`?

File: src/webaudio.js
Comment (thijstriemstra) [inline src/webaudio.js:307]: is this autoplay option necessary?

File: src/webaudio.js
Comment (kross-italk) [inline src/webaudio.js:307]: > is this autoplay option necessary? if so, add a comment why
It was already here, I do not think it's necessary since we're not invoking playback via `.play()`.

Comment (thijstriemstra) [reply src/webaudio.js:307]: ah yea missed that. I guess it doesn't hurt to leave it in for now.

**Review by thijstriemstra — APPROVED** [review]

File: src/webaudio.js
Comment (kross-italk) [reply src/webaudio.js:307]: @thijstriemstra Interesting. I tested with and without the autoplay. It no longer allocates the items in media-internals but doesn't playback any audio to the chosen sink. 
So now, I think a comment about this being load bearing is important.

Comment (thijstriemstra) [reply src/webaudio.js:307]: if you want to add another comment, sure, please do

Comment (thijstriemstra) [reply src/webaudio.js:307]: also, please make sure to test with firefox as well.

**Review by thijstriemstra — APPROVED** [review]

File: src/webaudio.js
Comment (kross-italk) [reply src/webaudio.js:307]: > also, please make sure to test with firefox as well.

I'm having trouble getting setSinkId/enumerateDevices to work in Firefox. and a harder time following the tracker. 
https://bugzilla.mozilla.org/show_bug.cgi?id=1610319
https://bugzilla.mozilla.org/show_bug.cgi?id=1493982

Comment (thijstriemstra) [reply src/webaudio.js:307]: you essentially didn't change the implementation, only the cleanup, and I tested it in Firefox (first) before so I can merge this and if you see any issues with Firefox, please open a ticket or new PR?

**Pull request merged**
